### PR TITLE
node: init blockchain state on startup

### LINKED ISSUE #3066: panic during object put
```
DsopKCVomH7wTFikghs1y9DtJhdW3P1BUJXQdnkBEbRL", "op": "PUT", "type": "peapod", "storage_id": "peapod"}
info	log/log.go:12	local object storage operation	{"shard_id": "AjU7Fguw6MmnYjhSYr6G4j", "address": "9Df2VFJBady7bRFtndSVs8J1AEd7o6QH43Rg2ckja3KH/DsopKCVomH7wTFikghs1y9DtJhdW3P1BUJXQdnkBEbRL", "op": "metabase PUT"}
debug	session/executor.go:30	serving request...	{"component": "SessionService", "request": "Create"}
panic: runtime error: integer divide by zero

goroutine 4840 [running]:
github.com/nspcc-dev/neofs-node/pkg/services/object/put.(*distributedTarget).Close(0xc00153b500)
	github.com/nspcc-dev/neofs-node/pkg/services/object/put/distributed.go:154 +0xd25
github.com/nspcc-dev/neofs-node/pkg/services/object/put.(*readyObjectPayloadWriter).Close(0x1e06970?)
	github.com/nspcc-dev/neofs-node/pkg/services/object/put/slice.go:121 +0x1b
github.com/nspcc-dev/neofs-sdk-go/object/slicer.writeInMemObject({0x1401190, 0xc003023b30}, {0x1401158, _}, {_, _}, {0xc00300ebd0, 0xc000634f80, 0xc000294ee0, {0x0, ...}}, ...)
	github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.12.0.20240809202351-256513c1b29b/object/slicer/slicer.go:656 +0x353
github.com/nspcc-dev/neofs-sdk-go/object/slicer.(*PayloadWriter)._writeChild(0xc00016f448, {0x1401190, 0xc003023b30}, {0x3e8, {0x14031c0, 0xc00073c480}, {0x1403200, 0xc001aea640}}, {0xc00081d520, 0x1, ...}, ...)
	github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.12.0.20240809202351-256513c1b29b/object/slicer/slicer.go:556 +0x326
github.com/nspcc-dev/neofs-sdk-go/object/slicer.(*PayloadWriter).writeLastChild(...)
	github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.12.0.20240809202351-256513c1b29b/object/slicer/slicer.go:508
github.com/nspcc-dev/neofs-sdk-go/object/slicer.(*PayloadWriter).Close(0xc00081d5d8?)
	github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.12.0.20240809202351-256513c1b29b/object/slicer/slicer.go:483 +0x30a
github.com/nspcc-dev/neofs-node/pkg/services/object/put.(*slicingTarget).Close(0xc001656e40)
	github.com/nspcc-dev/neofs-node/pkg/services/object/put/slice.go:87 +0x32
github.com/nspcc-dev/neofs-node/pkg/services/object/put.(*validatingTarget).Close(0xc0030239e0?)
	github.com/nspcc-dev/neofs-node/pkg/services/object/put/validation.go:147 +0x92
github.com/nspcc-dev/neofs-node/pkg/services/object/put.(*Streamer).Close(0xc001b701e0)
	github.com/nspcc-dev/neofs-node/pkg/services/object/put/streamer.go:269 +0x3e
github.com/nspcc-dev/neofs-node/pkg/services/object/put/v2.(*streamer).CloseAndRecv(0xc001648400)
	github.com/nspcc-dev/neofs-node/pkg/services/object/put/v2/streamer.go:109 +0x54
github.com/nspcc-dev/neofs-node/pkg/services/object/acl/v2.putStreamBasicChecker.CloseAndRecv(...)
	github.com/nspcc-dev/neofs-node/pkg/services/object/acl/v2/service.go:593
github.com/nspcc-dev/neofs-node/pkg/services/object.(*ResponseService).Put.func2()
	github.com/nspcc-dev/neofs-node/pkg/services/object/response.go:87 +0x1c
github.com/nspcc-dev/neofs-node/pkg/services/util/response.(*ClientMessageStreamer).CloseAndRecv(0xc001d7c9a8)
	github.com/nspcc-dev/neofs-node/pkg/services/util/response/client_stream.go:30 +0x22
github.com/nspcc-dev/neofs-node/pkg/services/object.(*putStreamResponser).CloseAndRecv(0xc000068a90)
	github.com/nspcc-dev/neofs-node/pkg/services/object/response.go:67 +0x25
github.com/nspcc-dev/neofs-node/pkg/services/object.(*SignService).Put.func2()
	github.com/nspcc-dev/neofs-node/pkg/services/object/sign.go:98 +0x1c
github.com/nspcc-dev/neofs-node/pkg/services/util.(*RequestMessageStreamer).CloseAndRecv(0xc003023b90)
	github.com/nspcc-dev/neofs-node/pkg/services/util/sign.go:90 +0x38
github.com/nspcc-dev/neofs-node/pkg/services/object.(*putStreamSigner).CloseAndRecv(0x13edf20?)
	github.com/nspcc-dev/neofs-node/pkg/services/object/sign.go:78 +0x1a
github.com/nspcc-dev/neofs-node/pkg/services/object.putStreamMetric.CloseAndRecv({{0x13fa4d8, 0xc000068a98}, {0x140eab8, 0xc000322508}, {0xc1d28e269b04f4d3, 0x108cae82f0, 0x1e2b260}})
	github.com/nspcc-dev/neofs-node/pkg/services/object/metrics.go:162 +0x45
github.com/nspcc-dev/neofs-node/pkg/network/transport/object/grpc.(*Server).Put(0xc0001d3310, {0x1408ce0, 0xc000046610})
	github.com/nspcc-dev/neofs-node/pkg/network/transport/object/grpc/service.go:80 +0x194
github.com/nspcc-dev/neofs-api-go/v2/object/grpc._ObjectService_Put_Handler({0x10f6ca0?, 0xc0001d3310}, {0x14058b8, 0xc00040c870})
	github.com/nspcc-dev/neofs-api-go/v2@v2.14.1-0.20240827150555-5ce597aa14ea/object/grpc/service_grpc.pb.go:714 +0xd8
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00013f200, {0x1401190, 0xc003023a40}, {0x1409e00, 0xc0015ee180}, 0xc00393e6c0, 0xc00160baa0, 0x1e19b20, 0x0)
	google.golang.org/grpc@v1.64.1/server.go:1673 +0x1208
google.golang.org/grpc.(*Server).handleStream(0xc00013f200, {0x1409e00, 0xc0015ee180}, 0xc00393e6c0)
	google.golang.org/grpc@v1.64.1/server.go:1794 +0xe3a
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.64.1/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 4850
	google.golang.org/grpc@v1.64.1/server.go:1040 +0x125
```

### PR DESCRIPTION
**Pull request by carpawell**
Fixes #3066 on cold starts.

### CODE REVIEW

**Review by cthulhu-rider — COMMENTED** [review]
although proposed change is rly needed, it does not cover division by zero totally. Zero can still come from the service pov. I suggest:
1. preserve zero return in implementations
2. throw more clear panic in the service

we can also consider defaulting

**Comment by cthulhu-rider** [comment]
@carpawell https://github.com/nspcc-dev/neofs-node/issues/3066#issuecomment-2561147763 what did u mean to fix?

**Comment by roman-khimov** [comment]
Epoch duration can't be 0. But this can an operator error. In which case panic is not the best way to handle the problem. I'd suggest relying on proper value everywhere, but check it on update (including node start) and using the default if it's wrong.

Also, looks like `t.objSharedMeta` calculations can be omitted `if !(t.localNodeInContainer && t.metainfoConsistencyAttr != "")`.

**Comment by cthulhu-rider** [comment]
https://github.com/nspcc-dev/neofs-node/blob/90194937dfe9c063650434ff27d291b95a5ebe27/pkg/network/transport/object/grpc/replication.go#L234 panics for the same reason

**Comment by carpawell** [comment]
> @carpawell https://github.com/nspcc-dev/neofs-node/issues/3066#issuecomment-2561147763 what did u mean to fix?

@cthulhu-rider, i meant do not encode meta info if it is not requested. It does not relate to panic directly so i would like to put this fix to another PR (like https://github.com/nspcc-dev/neofs-node/pull/3063 with some title change).

**Comment by carpawell** [comment]
> Also, looks like t.objSharedMeta calculations can be omitted if !(t.localNodeInContainer && t.metainfoConsistencyAttr != "").

@roman-khimov, agree, will fix but IMO not in a fast panic fix PR.

**Comment by cthulhu-rider** [comment]
> > panics for the same reason
> 
> @cthulhu-rider, what do you mean? It shares the same netmap state that is updated right after creation after this PR.

just a remark, i faced this when hotfixed original panic with default to get working tests. Anyway, these are different places which can potentially panic

File: cmd/neofs-node/reputation.go [hunk 1]
```diff
@@ -230,6 +230,12 @@ func initReputationService(c *cfg) {
 				return
 			}
 
+			if duration == 0 {
+				c.log.Warn("zero epoch duration received, fallback to default value",
```
Comment (cthulhu-rider) [inline cmd/neofs-node/reputation.go:234]: u can reuse `log` with embedded `epoch`

File: cmd/neofs-node/netmap.go [hunk 2]
```diff
@@ -22,6 +22,11 @@ import (
 	"go.uber.org/zap"
 )
 
+// defaultEpochDuration sets default epoch duration. Should not be used in a
+// normal situations but if no info can be fetched, smth should be used as a
```
Comment (cthulhu-rider) [inline cmd/neofs-node/netmap.go:26]: `if no info can be fetched, smth should be used` sounds like a philosophy to me. Simply `defaultEpochDuration is a default epoch duration to replace zero from FS chain.`

**Review by cthulhu-rider — COMMENTED** [review]
`node/nemtap:`

and i still recommend to throw particular panic in places where the original one could occur

File: cmd/neofs-node/netmap.go [hunk 3]
```diff
@@ -22,6 +22,11 @@ import (
...
+// fallback value.
+const defaultEpochDuration = 240
```
Comment (roman-khimov) [inline cmd/neofs-node/netmap.go:28]: IIRC we have this const in IR somewhere as well.

File: cmd/neofs-node/config.go [hunk 4]
```diff
@@ -743,6 +743,11 @@ func initBasics(c *cfg, key *keys.PrivateKey, stateStorage *state.PersistentStor
 
 	eDuration, err := nmWrap.EpochDuration()
 	fatalOnErr(err)
+	if eDuration == 0 {
```
Comment (roman-khimov) [inline cmd/neofs-node/config.go:746]: It's all in `neofs-node`, likely you can have some `nState.UpdateEpochDuration()` that will handle this.

**Comment by roman-khimov** [comment]
> throw particular panic in places where the original one could occur

It's like adding dead code, not worth the trouble. This panic is rather obvious one even when the only thing you see is line number and "division by zero".

File: cmd/neofs-node/config.go (hunk 4 above)
Comment (carpawell) [reply cmd/neofs-node/config.go]: ok, degraded logs a little then

File: cmd/neofs-node/netmap.go (hunk 3 above)
Comment (carpawell) [reply cmd/neofs-node/netmap.go:26]: tried to find (looked for `240`) but no, just magic numbers in neofs-adm, IR's deploy procedure, etc. also, i am not sure where it should be placed and if we should have a single const for every application

File: cmd/neofs-node/netmap.go (hunk 2 above)
Comment (carpawell) [reply cmd/neofs-node/netmap.go]: not a philosophy at all to me. my version explains when it is used and states that zero value is not ok, your version repeats `if` statement when this const is used. fixed

File: cmd/neofs-node/netmap.go [hunk 5]
```diff
@@ -62,6 +68,18 @@ func (s *networkState) setCurrentEpoch(v uint64) {
...
+func (s *networkState) updateEpochDuration(v uint64) {
+	if v != 0 {
+		s.epochDuration.Store(v)
+		return
+	}
+
+	s.l.Warn("zero epoch duration received, fallback to default value", zap.Uint64("applied default value", defaultEpochDuration))
+	s.epochDuration.Store(defaultEpochDuration)
+
+	return
```
Comment (cthulhu-rider) [inline cmd/neofs-node/netmap.go:80]: not needed

**Review by cthulhu-rider — APPROVED** [review]
dont forget about linter

**Comment by carpawell** [comment]
> It's like adding dead code

Same opinion, zero division is a funny cult mistake, in go IMO not needed any context, devs will always understand what it is about, admins can do nothing about it

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**